shacl12-core/index.html

```diff
@@ -7373,6 +7373,25 @@ <h3>sh:name and sh:description</h3>
 					multiple <a>values</a>, but should only have one <a>value</a> per language tag.
 				</p>
 			</section>
+			<section id="codeIdentifier">
+				<h3>sh:codeIdentifier</h3>
+				<p>
+					Shapes may have one <a>value</a> for <code>sh:codeIdentifier</code>
+					to suggest a name that can be used for a representation of the shape in APIs, query languages
+					and similar programmatic access.
+					The values of <code>sh:codeIdentifier</code> are <a>literals</a> with <a>datatype</a> <code>xsd:string</code>
+					where the string matches the regular expression <code>^[a-zA-Z_][a-zA-Z0-9_]*$</code>.
```

Comment:
Would this be better as a advice rather than a requirement? Some uses may require other characters.

Maybe put with the discussion note below.

Reply:
Does anyone know of a target language where the given (Java) naming convention would produce invalid identifiers? Because I think we should go with the lowest common denominator, i.e. some syntax that works for every language.

If there is a need for different syntaxes, a fresh property can be introduced, e.g. ex:javaScriptIdentifier could allow arbitrary strings.